src/main/java/com/empowerops/linqalike/CommonDelegates.java

```diff
@@ -32,7 +33,7 @@ public boolean equals(Object left, Object right) {
 
                 @Override
                 public int hashCode(Object object) {
-                    counter = (counter + 1) % Integer.MAX_VALUE; //mod prevents integer overflow
+                    counter = (counter % Integer.MAX_VALUE) + 1; //mod prevents integer overflow
```

Comment:
I should write a test to cover this....

and actually I got a little over eager, that original code would solve the problem iirc.